server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -225,11 +241,19 @@ ActionListener<SearchResponse> transformResponseListener(
             final SearchResponseProcessor processor = searchResponseProcessors.get(i);
 
             responseListener = ActionListener.wrap(r -> {
+                ProcessorExecutionDetail detail = new ProcessorExecutionDetail(processor.getType());
+                detail.addInput(Arrays.asList(r.getHits().deepCopy().getHits()));
```

Comment:
we need to spend cycles on doing shallow copy only in case verbose flag is set, please consider making corresponding changes